tests/EdgeToEdge/Routes/HandlePayPalPaymentNotificationRouteTest.php

```diff
@@ -20,22 +29,29 @@ class HandlePayPalPaymentNotificationRouteTest extends WebRouteTestCase {
 
 	const BASE_URL = 'https://that.paymentprovider.com/';
 	const EMAIL_ADDRESS = 'foerderpp@wikimedia.de';
+	const ITEM_NAME = 'My preciousss';
+	const UPDATE_TOKEN = 'my_secret_token';
 
 	public function testGivenValidRequest_applicationIndicatesSuccess() {
 		$this->createEnvironment( [], function ( Client $client, FunFunFactory $factory ) {
+			$factory->getDonationRepository()->storeDonation( ValidDonation::newIncompletePayPalDonation() );
+			$authorizer = new DoctrineDonationAuthorizationUpdater( $factory->getEntityManager() );
+			$authorizer->allowModificationViaToken(
+				1,
+				self::UPDATE_TOKEN,
+				DateTime::createFromFormat( 'Y-m-d H:i:s', '2039-12-31 23:59:59' )
+			);
 			$factory->setPayPalPaymentNotificationVerifier( $this->newNotifierMock() );
 
 			$client->request(
 				'POST',
 				'/handle-paypal-payment-notification',
-				[
-					'receiver_email' => self::EMAIL_ADDRESS,
-					'payment_status' => 'Completed'
-				]
+				$this->newRequest()
 			);
 
-			$this->assertSame( 'TODO', $client->getResponse()->getContent() );
 			$this->assertSame( 200, $client->getResponse()->getStatusCode() );
+			$donation = $factory->getDonationRepository()->getDonationById( 1 );
+			$this->assertDataComplemented( $donation, $this->newRequest() );
```

Comment:
It's odd to add new (logical) assertions to a test. Basically means you are testing two different things with one test.